bus-jmh/src/main/java/net/neoforged/bus/benchmarks/ManyBussesBenchmark.java

```diff
@@ -0,0 +1,46 @@
+package net.neoforged.bus.benchmarks;
+
+import net.neoforged.bus.api.*;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+@State(Scope.Benchmark)
+public class ManyBussesBenchmark {
+    private static final int BUS_COUNT = 100;
+
+    private static class TestEvent extends Event {
+        int value = 0;
+    }
+
+    private final IEventBus[] busses = new IEventBus[BUS_COUNT];
+
+    @Setup
+    public void setup() {
+        // Imitate FML-like setup where most mod event busses have few listeners for a given event.
+        for (int i = 0; i < BUS_COUNT; ++i) {
+            busses[i] = BusBuilder.builder().build();
+            if (i % 5 == 0) {
+                busses[i].addListener(TestEvent.class, event -> event.value++);
+            }
+            if (i % 7 == 0) {
+                busses[i].addListener(EventPriority.HIGH, TestEvent.class, event -> event.value++);
+            }
+            if (i % 9 == 0) {
+                busses[i].addListener(EventPriority.LOW, TestEvent.class, event -> event.value++);
+            }
+        }
+    }
+
+    @Benchmark
+    public int testManyBusses() {
```

Comment:
Could we potentially have another benchmark here which posts to each event via the `post(Event)` call, for comparison's sake?

Reply:
There you go:

```
Benchmark                                          Mode  Cnt      Score      Error  Units
ManyBussesBenchmark.testManyBusses                 avgt   15   1115.624 ▒   29.924  ns/op
ManyBussesBenchmark.testManyBussesPerPhase         avgt   15  16236.739 ▒ 1644.855  ns/op
```
(first one is `post(Event)`)